tests/submodule/add.c

```diff
@@ -128,3 +128,51 @@ void test_submodule_add__url_relative_to_workdir(void)
 
 	assert_submodule_url("TestGitRepository", git_repository_workdir(g_repo));
 }
+
+void test_submodule_add__path_exists_in_index(void)
+{
+	git_index *index;
+	git_submodule *sm;
+	git_buf dirname = GIT_BUF_INIT;
+	git_buf filename = GIT_BUF_INIT;
+
+	/* In this repo, HEAD (master) has no remote tracking branc h*/
```

Comment:
There's a typo. Also: why does this matter for our particular unit test here? I don't want to propose it doesn't, but for myself I don't get why this is interesting. Please either extend that comment and explain a bit more why this is important here or just remove it :)